platform-hybrid-service-framework/src/main/java/org/hypertrace/core/serviceframework/hybrid/HybridPlatformService.java

```diff
@@ -37,7 +38,13 @@ protected void doStop() {
 
   protected abstract List<GrpcPlatformServerDefinition> getServerDefinitions();
 
-  protected abstract List<HttpHandlerFactory> getHandlerFactories();
+  protected List<HttpHandlerFactory> getHttpHandlerFactories() {
+    return List.of();
+  }
+
+  protected List<HybridHttpHandlerFactory> getHybridHttpHandlerFactories() {
```

Comment:
The extra interface is a variant of the existing HttpHandlerFactory that gives the construction context access to the GRPC environment as well. The handlers the factory returns are no different, but these extra arguments are only available to hybrid containers. This is typically used for an HTTP handler to then turn around and make a call to the GRPC container.